app/controllers/workflows_controller.rb

```diff
@@ -1,15 +1,114 @@
 # frozen_string_literal: true
 
 class WorkflowsController < Sellers::BaseController
-  before_action :set_body_id_as_app
+  before_action :set_workflow, only: %i[edit update destroy]
+  before_action :authorize_workflow, only: %i[edit update destroy]
+  before_action :fetch_product_and_enforce_ownership, only: %i[create update], if: -> { product_or_variant_workflow? }
+
+  layout "inertia"
+
+  FLASH_CHANGES_SAVED = "Changes saved!"
+  FLASH_WORKFLOW_PUBLISHED = "Workflow published!"
+  FLASH_WORKFLOW_UNPUBLISHED = "Unpublished!"
+  FLASH_WORKFLOW_DELETED = "Workflow deleted!"
 
   def index
     authorize Workflow
     create_user_event("workflows_view")
+
+    workflows_presenter = WorkflowsPresenter.new(seller: current_seller)
+    render inertia: "Workflows/Index", props: workflows_presenter.workflows_props
+  end
+
+  def new
+    authorize Workflow
+    workflow_presenter = WorkflowPresenter.new(seller: current_seller)
+    render inertia: "Workflows/New", props: {
+      context: -> { workflow_presenter.workflow_form_context_props }
+    }
```

Comment:
using lambda's here in this file so every prop does not get loaded on every request, thus reducing the payload and increasing the API response time